include/dds/core/xtypes/detail/Annotation.hpp

```diff
@@ -17,27 +17,161 @@
 
 #ifndef EPROSIMA_DDS_CORE_XTYPES_DETAIL_ANNOTATION_HPP_
 #define EPROSIMA_DDS_CORE_XTYPES_DETAIL_ANNOTATION_HPP_
+#include <stdint.h>
+#include <string>
+#include <dds/core/xtypes/AnnotKind.hpp>
```

Comment:
Please, correct me if I am wrong. I see the detail package as a monolithic lib that doesn't know anything about the no-detail packages.